onnxscript/optimizer/_constant_folding.py

```diff
@@ -300,6 +340,54 @@ def _get_int_attribute(node: ir.Node, name: str, default: int | None = None) ->
     return default
 
 
+@register("Abs")
+def abs(node: ir.Node, op, state: OptimizerState) -> ReturnValue:
+    """Replace an Abs node by Identity when applicable.
+
+    Currently, addresses Abs applied to symbolic shapes.
+    """
+    input = _get_input(node, 0)
+    input_sym_value = state.get_shape_value(input)
```

Comment:
Do we need to ensure that input is the output of Shape? Or it this guaranteed / unrelated?

Reply:
This is relying on the invariant maintained by the optimizer that any "symbolic" name present in the returned value is a real dimension-value obtained via a Shape op.

Reply:
If we need to generalize this in the future, we will need to explicitly add another boolean flag to track whether a symbolic name is known to be non-negative.